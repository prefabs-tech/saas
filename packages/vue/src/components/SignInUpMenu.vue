<template>
  <ResponsiveMenu :routes="routes" />
</template>

<script lang="ts">
export default {
  name: "SignInUpMenu",
};
</script>

<script setup lang="ts">
import { useConfig } from "@dzangolab/vue3-config";
import { useI18n } from "@dzangolab/vue3-i18n";
import { ResponsiveMenu } from "@dzangolab/vue3-ui";
import { computed } from "vue";

import { useTranslations } from "../index";

import type { AppConfig } from "@dzangolab/vue3-config";

const config = useConfig() as AppConfig;

const messages = useTranslations();

const { t } = useI18n({ messages });

const signUpRoute = {
  name: t("user.menu.signup"),
  route: "signup",
};

const loginRoute = {
  name: t("user.menu.signin"),
  route: "login",
};

const routes = computed(() => {
  if (config?.user?.routes?.signup.disabled) {
    return [loginRoute];
  }

  return [loginRoute, signUpRoute];
});
</script>
