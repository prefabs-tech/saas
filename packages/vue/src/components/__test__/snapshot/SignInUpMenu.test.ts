import configPlugin from "@dzangolab/vue3-config";
import i18nPlugin, { useLocaleStore } from "@dzangolab/vue3-i18n";
import { mount, RouterLinkStub } from "@vue/test-utils";
import { createPinia } from "pinia";
import { describe, it, expect } from "vitest";

import SignInUpMenu from "@/components/SignInUpMenu.vue";

import appConfig from "../config";

import type { VueWrapper } from "@vue/test-utils";

describe("SignInUpMenu", () => {
  const pinia = createPinia();

  const { setLocale } = useLocaleStore(appConfig.slug);

  const locales = ["en", "fr"];

  for (const locale of locales) {
    setLocale(locale);

    const wrapper: VueWrapper = mount(SignInUpMenu, {
      global: {
        plugins: [
          pinia,
          [
            configPlugin,
            {
              config: appConfig,
            },
          ],
          [
            i18nPlugin,
            {
              config: appConfig,
            },
          ],
        ],
        stubs: {
          RouterLink: RouterLinkStub,
        },
      },
    });

    it("matches snapshot in " + locale, () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  }
});
