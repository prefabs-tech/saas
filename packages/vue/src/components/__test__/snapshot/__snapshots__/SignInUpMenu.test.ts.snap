// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignInUpMenu > matches snapshot in en 1`] = `
<nav
  class="menu"
>
  <ul>
    
    <li
      class=""
    >
      <a>
        Signin
      </a>
    </li>
    <li
      class=""
    >
      <a>
        Signup
      </a>
    </li>
    
  </ul>
</nav>
`;

exports[`SignInUpMenu > matches snapshot in fr 1`] = `
<nav
  class="menu"
>
  <ul>
    
    <li
      class=""
    >
      <a>
        Connexion
      </a>
    </li>
    <li
      class=""
    >
      <a>
        Inscription
      </a>
    </li>
    
  </ul>
</nav>
`;
